src/core/manager.c

```diff
@@ -1970,6 +1970,14 @@ int manager_load_unit_prepare(
         if (!name)
                 name = basename(path);
 
+        /* We might have been given a full path as a name by portablectl attach.
+         * Get the name out of it, allowing to load the unit bypassing daemon-reload. */
+        if (!path && is_path(name) &&
+                        (startswith(name, "/run/systemd/system.attached") || startswith(name, "/etc/systemd/system.attached"))) {
+                path = name;
+                name = basename(path);
+        }
+
```

Comment:
i don't like this kind of special casing at all. Isn't the better approach to use "--no-reload" on the client side when processing many portable images at once? and then fire off a "systemctl daemon-reload" only once at the very end?

Reply:
Yes, I do not like the special casing either, I added it to restrict the scope explicitly.

As mentioned in the description, sadly batching is not always an option - unfortunately I have little say in what the orchestrator does, and the platform has to adapt to it rather than the other way around. Also, even with a single service, having an additional delay of 30 seconds to a minute between deployment and service start due to the reload is enough to break things in qemu, where automated tests are run.

Would it be more acceptable without the special casing?

Thanks for the review!

Reply:
Another situation where running daemon-reload is an issue is during the boot process - in our case, the rootfs is read-only so portable services are attached runtime-only, and get re-attached on boot by an embedded service every time after the external, encrypted and network-unlocked storage becomes available. In my experience running daemon-reload during boot can cause hard to debug race conditions, and I'd really want to avoid it.